RD-5127 Fix test `deployment_button_configuration_spec.ts`

leon-good-life opened the pull request:
<!--- Provide JIRA issue ID and a general summary of your changes in the Title above -->
<!--- e.g. "RD-1234 Improve styling in SuperComponent" -->
<!--- Note that the title is used as a commit message after merging the PR -->

## Description
Jira ticket: https://cloudifysource.atlassian.net/browse/RD-5127
<!--- Describe your changes to help reviewers understand the details. -->

## Screenshots / Videos
N/A
<!--- If applicable, add screenshot(s) or video(s) describing the changes you made. -->
<!--- Consider adding comparison screenshot(s) - before and after the change. -->
<!--- If not applicable, just write “N/A” in this section. -->

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2981/pipeline
<!--- If applicable, describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
<!--- If applicable, provide a link to system tests, e.g: -->
<!--- [#1039](https://jenkins.cloudify.co/.../1039) (2021-08-06 07:05:50) --> 
<!--- If not applicable, just write “N/A” in this section. -->

## Documentation
N/A
<!--- If applicable, describe how you documented or plan to document your changes. -->
<!--- Check UI documentation guidelines - https://cloudifysource.atlassian.net/l/c/EYWJ0XfB - for details. -->
<!--- Provide links to JIRA issues, other PRs or related documentation pages. -->
<!--- If not applicable, just write “N/A” in this section. -->

leon-good-life commented:
I made an attempt and started running system tests:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2981/pipeline

leon-good-life commented:
I fixed the test, it passed. But I will try the suggestion of @qooban : `cy.setSearchableDropdownValue`.

leon-good-life commented:
I started the test again on Jenkins after using `cy.setSearchableDropdownValue`: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2982/pipeline

qooban commented on test/cypress/integration/widgets/deployment_button_configuration_spec.ts:
```diff
@@ -43,9 +43,12 @@ describe('Create deployment button widget should allow configuring', () => {
-                cy.contains(withoutLabelsBlueprint).should('exist');
                 cy.contains(labelsBlueprint).should('not.exist');
             });
+            cy.setSearchableDropdownValue('Blueprint', withoutLabelsBlueprint);
+            openDropdown('blueprintName').within(() => {
```
Does it work as expected? Seems to me like we are opening the dropdown twice, first in `cy.setSearchableDropdownValue` and second in `openDropdown` function.

The same comment about L68-69.

leon-good-life replied:
Yes, the dropdown is opening twice. The first line is filtering. The second line is verifying. I'm looking and thinking how can I merge those two lines. they are working. just to make the code look better.

leon-good-life replied:
I will try `cy.setSearchableDropdownValue('Blueprint', withoutLabelsBlueprint).within(  )`

leon-good-life replied:
@kubama good idea, so if the item does not exist when running `setSearchableDropdownValue` the test fails automatically, right? so I don't need to verify. 

But I think maybe it worth to me to add a single line comment that indicates that this line is verifies that an item exists.

kubama replied:
My understanding is that the value in the dropdown is selected as it is required by the test flow (or is it not?). If so the test should fail anyway if the selection is not successful. I don't see a reason to explicitly check if the selection succeeded. This is something that could be done in a dropdwon component test, not an integration test.

leon-good-life replied:
I think I addressed this comment.  Because I need to check two blueprints  I used `cy.get('input').type(value);`.

kubama replied:
Looks like I got confused and selecting the value is not actually required... if so then it's good that it was possible to remove it :+1:

leon-good-life commented:
I addressed review comment, I verified on my machine that tests correctly and passes, and I started the test on Jenkins: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2985/pipeline

leon-good-life commented:
The latest code is tested in https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2986/pipeline/   (it is `RD-5127-attempt` branch that I created. But the code is 100% the same.)

qooban commented on test/cypress/integration/widgets/deployment_button_configuration_spec.ts:
```diff
@@ -42,10 +42,13 @@ describe('Create deployment button widget should allow configuring', () => {
+            cy.getField('Blueprint')
+                .click()
+                .within(() => {
+                    cy.contains(labelsBlueprint).should('not.exist');
```
IIRC `DynamicDropdown` component lists only first 10 options on first open. There can be any number of blueprints uploaded in Cloudify Manager, so without limiting the list of options provided by the Blueprint dropdown we are not sure if the `labelsBlueprint` blueprint does not exist.

Maybe you could just add `cy.get('input').type(labelsBlueprint);` line before that check and adjust L49 to clear input before typing?

leon-good-life replied:
I addressed it now: [bd16692](https://github.com/cloudify-cosmo/cloudify-stage/pull/2018/commits/bd16692a29323a45f757d0facb927272095e19b3)

I started this test on Jenkins again: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2988/pipeline

leon-good-life replied:
The test passed
